internal/client/db/children.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package db
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/internal/primitives/core/hash"
+	"github.com/ChainSafe/gossamer/internal/primitives/database"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// Functionality for reading and storing children hashes from db.
+
+// Returns the hashes of the children blocks of the block with `parentHash`.
+func readChildren[H comparable](
+	db database.Database[hash.H256], column database.ColumnID, prefix []byte, parentHash H,
+) ([]H, error) {
+	encoded := scale.MustMarshal(parentHash)
+	key := append(prefix, encoded...)
+
+	rawValOpt := db.Get(column, key)
+
+	if rawValOpt == nil {
+		return nil, nil
+	}
+	rawVal := rawValOpt
```

Comment:
Raw value optional is not necessary
```suggestion
	rawVal := db.Get(column, key)
	if rawVal == nil {
		return nil, nil
	}
```